app/bundles/ChannelBundle/Model/MessageQueueModel.php

```diff
@@ -177,7 +176,6 @@ public function queue(
         if ($messageQueues) {
             $this->saveEntities($messageQueues);
             $messageQueueRepository = $this->getRepository();
-            assert($messageQueueRepository instanceof MessageQueueRepository);
```

Comment:
This is an example of where I think we should keep assert. `app/bundles/EmailBundle/Tests/Model/EmailModelTest.php)` is creating the wrong repo for the mock. This is caught by the assert.

Auto wiring is great but services can be overwritten in tests/plugins/bundles etc.. Asserts will help us enforce typing and catch issues better than phpstan IMO.

Reply:
@diaboloshogunate Well this one is tricky one, I saw the bug you are talking about and I [fixed it](49f8a5b7f838b1bc7768f59e968939ba46593d73), if we keep the `assert` stan will fail, if we remove it the test should fail but it does not, maybe because it is a Mock. 

I agree with you about autowiring, do you have suggestions to solve this puzzle?

Reply:
The link is broken. What is the error phpstan is giving if we keep this assert?

Reply:
Here is the fix 
https://github.com/mautic/mautic/pull/12262/commits/49f8a5b7f838b1bc7768f59e968939ba46593d73

Reply:
```
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   app/bundles/ChannelBundle/Model/MessageQueueModel.php                                                                                                       
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  179    Call to function assert() with true will always evaluate to true.                                                                                           
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your                                 
            phpstan.neon.                                                                                                                                            
  179    Instanceof between Mautic\ChannelBundle\Entity\MessageQueueRepository and Mautic\ChannelBundle\Entity\MessageQueueRepository will always evaluate to true.  
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your                                 
            phpstan.neon.                                                                                                                                            
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------ 
```

Reply:
@diaboloshogunate Let's keep this issue in our radar, I think the next PR we will have more strongly typed classes, I will mark this issue as resolved for now, but I will ask your revision on the next PR